### PR TITLE
SDCICD-187: Docs refresh

### PR DESCRIPTION
**Pull request by jeefy**
Updating a bunch of docs.

Things of note: 
- Removing Options.md as that doc is becoming unwieldy to maintain and essentially duplicates our comments in the actual `config.go` 
- Moving markdown links to the bottom of the doc. Also removes duplicate links :pray:
- Several fixed links

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -142,25 +144,34 @@ Collecting metrics of the above tests as well as testing recovery of failures. E
...
 3. Verify Cluster Integrity
 4. Run Tests (pre-upgrade)
-5. Capture / Upload logs to GCS
+5. Capture logs, metrics, and metadata
 6. Upgrade Cluster
 7. Verify Cluster Integrity
 8. Run Tests (post-upgrade)
-9. Capture / Upload logs to GCS
+9. Capture logs, metrics, and metadata 
+10. CI-Operator uploads artifacts to GCS
```
Comment (meowfaceman) [inline README.md:156]: This isn't part of things AFAIK -- however prow is configured just automagically does this.

File: docs/Addons.md [hunk 2]
```diff
@@ -30,16 +28,15 @@ env:
...
 
 Your account will need the following permissions:
 
 *   Credentials API access
-*   ...
 
 ### **Configuring your job to use your OCM refresh token**
 
-In order to run addon tests in osde2e, you will need to create a secret in Origin CI with your OCM refresh token. Please follow [these instructions](https://github.com/openshift/release/blob/e877df16a32be22b60a62b6313ef3e0fe2e9256b/core-services/secret-mirroring/README.md) to both create a secret and a secret mapping into the ci namespace.
+In order to run addon tests in osde2e, you will need to create a secret in Origin CI with your OCM refresh token. Please follow [these instructions] to both create a secret and a secret mapping into the ci namespace.
```
Comment (meowfaceman) [inline docs/Addons.md:39]: Let's take this opportunity to mention that the OCM refresh token needs to be named `ocm-refresh-token` as well.

File: docs/Addons.md [hunk 3]
```diff
@@ -50,7 +47,16 @@ cicd_event{environment="int",event="InstallSuccessful",install_version="openshif
 
 cicd_jUnitResult{environment="int",install_version="openshift-v4.2.0-0.nightly-2020-01-15-224532",job="periodic-ci-openshift-osde2e-master-e2e-int-4.2-4.2",monitor="datahub",phase="install",result="failed",suite="OSD e2e suite",testname="[OSD] Managed Velero Operator deployment should have all desired replicas ready",upgrade_version="openshift-v4.2.0-0.nightly-2020-01-15-231532"}
 
-cicd_metadata{cluster_id="1a2bc3",environment="int",install_version="openshift-v4.2.0-0.nightly-2020-01-15-224532",job="periodic-ci-openshift-osde2e-master-e2e-int-4.2-4.2",job_id="123",metadata_name="time-to-cluster-ready",monitor="datahub",phase="",upgrade_version="openshift-v4.2.0-0.nightly-2020-01-15-231532"}
+cicd_addon_metadata{cluster_id="1a2bc3",environment="int",install_version="openshift-v4.2.0-0.nightly-2020-01-15-224532",job="periodic-ci-openshift-osde2e-master-e2e-int-4.2-4.2",job_id="123",metadata_name="time-to-cluster-ready",monitor="datahub",phase="",upgrade_version="openshift-v4.2.0-0.nightly-2020-01-15-231532"}
```
Comment (meowfaceman) [inline docs/Addons.md:50]: Let's keep `cicd_metadata` and add in `cicd_addon_metadata`, as both are present. Let's also note (somewhere) that only numerical addon metadata is captured.

**Comment by jeefy** [comment]
@meowfaceman Nits addressed :+1: PTAL

**Pull request merged**